pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderToken.java

```diff
@@ -237,6 +280,69 @@ private static String validateToken(final String token) throws AuthenticationExc
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private DecodedJWT authenticateTokenJwk(final String token) throws AuthenticationException {
```

Comment:
Instead of adding this code to the AuthenticationProvider could we refactor some of this code out into separate classes so there is not a lot of if else checks (i think that makes the code more difficult to follow test and maintain) within the code?

Maybe there can be a TokenDecoder interface and TokenInfo or Token class. Then there can be multiple implementations of the TokenDecoder and the code can be simplified and not care how it works underneath. The TokenInfo or Token class can add more attributes over time but at this point I think we only care about the principal and expiration.

Thoughts?

Reply:
Sounds like a good idea 👍 

I'll see what I can do 🔍

Reply:
@cckellogg I added the feature as a separate plugin instead. This gets rid of most of the if else statements and the jwkEnable flag which makes it a bit cleaner 😄